tilequeue/queue/multisqs.py

```
@@ -0,0 +1,181 @@
+                if len(queue_buf) == self.queue_buf_size:
+                    sqs_queue = self.sqs_queue_for_name.get(sqs_queue_name)
+                    assert sqs_queue_name, \
+                        'Missing queue for: %s' % sqs_queue_name
+                    self._write_batch(sqs_queue, queue_buf)
+                    del queue_buf[:]
+
+        for queue_name, queue_buf in buf_per_queue.items():
+            if queue_buf:
+                sqs_queue = self.sqs_queue_for_name.get(queue_name)
+                assert sqs_queue, 'Missing queue for: %s' % queue_name
+                self._write_batch(sqs_queue, queue_buf)
+
+        return n_queued, n_in_flight
+
+    def read(self, max_to_read=None):
+        if max_to_read is None:
+            max_to_read = self.queue_buf_size
+
+        coord_messages = []
+        left_to_read = max_to_read
+
+        for sqs_queue in self.sqs_queues:
+
+            queue_messages = sqs_queue.get_messages(
+                num_messages=left_to_read,
+                attributes=('SentTimestamp',))
+
+            for qm in queue_messages:
+
+                data = qm.get_body()
+                coord = deserialize_coord(data)
+                if coord is None:
+                    # TODO log?
+                    continue
+
+                try:
+                    timestamp = float(qm.attributes.get('SentTimestamp'))
+                except (TypeError, ValueError):
+                    timestamp = None
+
+                coord_message = CoordMessage(coord, qm, timestamp)
+                coord_messages.append(coord_message)
+                left_to_read -= 1
+
+            assert(left_to_read >= 0)
+            if left_to_read == 0:
+                break
+
+        return coord_messages
+
+    def job_done(self, coord_message):
+        queue_name = self.get_queue_name_for_zoom(coord_message.coord.zoom)
+        if queue_name is None:
+            # TODO log? should this assert instead?
```

Added assert in 22ea268.

And upon further thought, I think that you're right that we'd want it to fail fast when it didn't find a queue for a particular input. That suggests either the mapping is wrong/incomplete, or the input coordinate is invalid, which I think we'd want to know about right away. I'll go ahead and make this change.